File: src/main/java/com/indiepost/service/PostUserInteractionService.java
package com.indiepost.service;

import com.indiepost.dto.post.PostUserInteraction;
import com.indiepost.model.Bookmark;
import com.indiepost.model.PostReading;

public interface PostUserInteractionService {

    Long add(Long userId, Long postId);

    PostReading findOne(Long id);

    PostUserInteraction findUsersByPostId(Long postId);

    PostReading findOneByPostId(Long postId);

    Bookmark findBookmark(Long userId, Long postId);

    void setInvisible(Long postId);

    void setInvisibleAll();

    void setVisibleAll();

    void addBookmark(Long postId);

    void removeBookmark(Long postId);

    void removeAllUsersBookmarks();

    void deleteById(Long id);
}

File: src/main/kotlin/com/indiepost/service/PostUserInteractionService.kt
package com.indiepost.service

import com.indiepost.dto.post.PostUserInteraction
import com.indiepost.model.Bookmark
import com.indiepost.model.PostReading

interface PostUserInteractionService {

    fun add(userId: Long, postId: Long): Long

    fun findOne(id: Long): PostReading?

    fun findUsersByPostId(postId: Long): PostUserInteraction?

    fun findOneByPostId(postId: Long): PostReading?

    fun findBookmark(userId: Long, postId: Long): Bookmark?

    fun setInvisible(postId: Long)

    fun setInvisibleAll()

    fun setVisibleAll()

    fun addBookmark(postId: Long)

    fun removeBookmark(postId: Long)

    fun removeAllUsersBookmarks()

    fun deleteById(id: Long)
}
